backend/routes/session.ts

```diff
@@ -1,38 +1,46 @@
 import { Request, Response, Router } from 'express';
-import createVideoService from '../videoServiceFactory';
+import createVideoService from '../videoService/videoServiceFactory';
 import getSessionStorageService from '../sessionStorageService';
+import blockCallsForArgs from '../helpers/blockCallsForArgs';
 
 const sessionRouter = Router();
 const videoService = createVideoService();
 const sessionService = getSessionStorageService();
 
-sessionRouter.get('/:room', async (req: Request, res: Response) => {
+/**
+ * Gets or creates a session in a thread safe (async operation safe) way.
+ * The function is wrapped in blockCallsForArgs to ensure that simultaneous calls for a room name return the same session.
+ * Not blocking these calls would result in separate sessions being created for the same room name.
+ * @param {string} roomName - name of the meeting room
+ * @returns {Promise<string>} the sessionId
+ */
+export const getOrCreateSession = blockCallsForArgs(async (roomName: string) => {
```

Comment:
Is it worth moving this to its own helper file as a higher-order function/factory to tidy up the `session.ts` file?

Reply:
I could do, that would probably simplify the test file too with some injection of the services